src/components/base/outline-button/outline-button.ts

```diff
@@ -51,7 +55,16 @@ export class OutlineButton extends OutlineElement {
   /**
    * The button style variant to use.
    */
-  @property({ type: String, attribute: 'button-variant' })
+  @property({
+    type: String,
+    attribute: 'button-variant',
+    converter: value => {
```

Comment:
Adding a property converter to check if the provided value is in our custom type